Parser/tokenizer.c

```diff
@@ -1330,6 +1329,11 @@ parser_warn(struct tok_state *tok, PyObject *category, const char *format, ...)
 static int
 warn_invalid_escape_sequence(struct tok_state *tok, const char first_invalid_escape_char)
 {
+
+    if (!tok->tok_report_warnings) {
```

Comment:
There is no other way that I can think of handling this elegantly. The second time we recurse we do it because raising the warning needs to know the encoding and that creates a new clean tokenizer. So unless we stick something in the interpreter state (yuk!) the only way is that the function to get encoding just says "no warning thanks".

Reply:
Was trying to figure out whether we can do something else, but, yeah, this seems like the only way.